foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/queries/DatasourceCallQueryMechanism.java

```diff
@@ -379,8 +381,8 @@ public void insertObject() throws DatabaseException {
             shouldAcquireValueAfterInsert = descriptor.getSequence().shouldAcquireValueAfterInsert();
         }
         Collection returnFields = null;
-        if (descriptor.hasReturningPolicy()) {
-            returnFields = descriptor.getReturningPolicy().getFieldsToMergeInsert();
+        if (descriptor.hasReturningPolicy() && descriptor.getReturnFieldsToMergeInsert() != null) {
```

Comment:
Don't really need to check for returningPolicy anymore as we did that during descriptor init

Reply:
Fixed